### PR TITLE
docs: improve the auth state docs

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs/src/test-auth-js.md [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+  // page is signed in.
+});
+
+test('second', async ({ page }) => {
+  // page is signed in.
+});
+```
+
+```js js-flavor=js
+import { test } from '@playwright/test';
```
Comment (dgozman) [inline docs/src/test-auth-js.md:42]: This snippet is ts, and previous is js, but they are marked as opposite.

File: docs/src/test-auth-js.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+import { chromium, FullConfig } from '@playwright/test';
+
+async function globalSetup(config: FullConfig) {
+  const browser = await chromium.launch();
+  const page = await browser.newPage();
+  await page.goto('https://github.com/login');
+  await page.fill('input[name="user"]', 'user');
+  await page.fill('input[name="password"]', 'password');
+  await page.click('text=Sign in');
+    // Save signed-in state to 'storageState.json'.
```
Comment (dgozman) [inline docs/src/test-auth-js.md:99]: Indentation is off, and comment is missing in js snippet.

File: docs/src/test-auth-js.md [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+:::note
+If you can log in once and commit the `storageState.json` into the repository, you won't need the global
+setup at all, just specify the `storageState.json` in Playwright Config as above and it'll be picked up.
+:::
+
+### Multiple signed in roles
+
+Sometimes you have more than one signed-in user in your end to end tests. You can achieve that via logging in for these users multiple times in globalSetup and saving that state into different files.
+
+```js js-flavor=js
```
Comment (dgozman) [inline docs/src/test-auth-js.md:164]: This snippet is stale :)

File: docs/src/test-auth-js.md [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+    // page is signed in as a user.
+  });
+});
+```
+
+### Reuse the signed in page in multiple tests
+
+Although discouraged, sometimes it is necessary to sacrifice the isolation and run a number of tests
+in the same page. In that case, you can log into that page once in `beforeAll` and then use that same
+page in all the tests. Note that you need to run these tests serially using `test.describe.serial` in
```
Comment (dgozman) [inline docs/src/test-auth-js.md:241]: Let's link to the serial/page reuse doc?

**Review by dgozman — APPROVED** [review]

File: docs/src/test-auth-js.md (hunk 4 above)
Comment (pavelfeldman) [reply docs/src/test-auth-js.md:241]: Linking to retries feels artificial here, we should re-create core concepts and link there.

**Pull request merged**